### PR TITLE
[13.0][IMP] dms: Portal

### PR DESCRIPTION
**Pull request by victoralmau**
**Portal functionallity**

Users with portal access view link to "Documents"
In Documents show all "root" directories with explicit access to portal and files it.

This new feature use Access Group field in directories and files. It's neccessary "read" permission to show in portal.

If directory have access group to portal, by default, all subdirectories too, but if "not inherit groups" field not selected dissapear.

Add too "Share" link to Directories and Files to show it with custom link

dms_1
dms_2
dms_3
dms_share

Please @pedrobaeza review it

V12: https://github.com/OCA/dms/pull/22

@Tecnativa TT25647

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Trying a share link in portal with anonymous user, I get an internal server error:

http://3442854-25-5e0bd1.runbot2-3.odoo-community.org/mail/view?model=dms.directory&res_id=1&access_token=227726a4-da82-49a1-a9da-f3fc915854b2

**Comment by victoralmau** [comment]
> Trying a share link in portal with anonymous user, I get an internal server error:
> 
> http://3442854-25-5e0bd1.runbot2-3.odoo-community.org/mail/view?model=dms.directory&res_id=1&access_token=227726a4-da82-49a1-a9da-f3fc915854b2

I checking too in V12 and _share_link_ in `dms_directory` isn't correct (i will open PR to fix it now) > https://github.com/OCA/dms/pull/26

Related to error, in V12, auto-redirect `@http.route` (anonymus user) to login, how is it done in v12?

**Comment by victoralmau** [comment]
> Trying a share link in portal with anonymous user, I get an internal server error:
> 
> http://3442854-25-5e0bd1.runbot2-3.odoo-community.org/mail/view?model=dms.directory&res_id=1&access_token=227726a4-da82-49a1-a9da-f3fc915854b2

I had checked redirect again in runbot and work fine.

**Comment by pedrobaeza** [comment]
Please try on an incognito window.

**Comment by victoralmau** [comment]
> Please try on an incognito window.

Please, check now

File: dms/controllers/portal.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+from odoo.osv.expression import OR
+
+from odoo.addons.portal.controllers.portal import CustomerPortal
+from odoo.addons.web.controllers.main import content_disposition, ensure_db
+
+_logger = logging.getLogger(__name__)
+
+
+class CustomerPortal(CustomerPortal):
+    def _dms_check_access(self, model, res_id, access_token=None):
```
Comment (pedrobaeza) [inline dms/controllers/portal.py:17]: This doesn't contain latest change about this part AFAIK.

Comment (victoralmau) [reply dms/controllers/portal.py:17]: In this PR appear all the changes related to FIX share option https://github.com/OCA/dms/pull/26 
(https://github.com/OCA/dms/pull/22 + https://github.com/OCA/dms/pull/26)

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Let's merge it as this is basically the same as in v12:

/ocabot merge major

**Pull request merged**